typed-racket-lib/typed-racket/typecheck/tc-structs.rkt

```diff
@@ -311,7 +337,8 @@
   (define parent-tys (map fld-t (get-flds parent-type)))
 
   (define names (get-struct-names nm fld-names #f))
-  (define desc (struct-desc parent-tys tys null #t #f))
+  ;; built-in structs are assumed to be immutable with immutable parents
+  (define desc (struct-desc parent-tys tys null #f #f #f))
```

Comment:
Ok, I'll assume that the built-in structs aren't going to grow some mutable data.